ARCs/arc-draft_extended_desc.md

```diff
@@ -0,0 +1,160 @@
+status: Draft
+type: Standards Track
+category: ARC
+created: 2023-11-14
+requires: 4
+---
+
+
+## Abstract
+This ARC takes the existing JSON description of a contract as described in ARC4 and adds more fields for the purpose of client interaction
+
+## Motivation
+The description provided by ARC4 is missing a lot of critical information that clients should know when interaction with an app.
+
+## Specification
+```ts
+/** Mapping of named structs to the ABI type of their fields */
+interface StructFields {
+  [name: string]: string | StructFields;
+}
+
+/** Describes a single key in app storage */
+interface StorageKey {
+  /** The type of the key. Can be ABI type or named struct */
+  keyType: string;
+  /** The type of the value. Can be ABI type or named struct */
+  valueType: string;
+  /** The key itself, as a byte array */
+  key: number[];
+}
+
+interface StorageMap {
+  /** The type of the key. Can be ABI type or named struct */
+  keyType: string;
+  /** The type of the value. Can be ABI type or named struct */
+  valueType: string;
+  /** The prefix of the map, as a string */
+  prefix: string;
+}
+
+interface Method {
+  /** The name of the method */
+  name: string;
+  /** Optional, user-friendly description for the method */
+  desc?: string;
+  /** The arguments of the method, in order */
+  args: Array<{
+    /** The type of the argument */
+    type: string;
+    /** If the type is a struct, the name of the struct */
+    struct?: string;
+    /** Optional, user-friendly name for the argument */
+    name?: string;
+    /** Optional, user-friendly description for the argument */
+    desc?: string;
+  }>;
+  /** Information about the method's return value */
+  returns: {
+    /** The type of the return value, or "void" to indicate no return value. */
+    type: string;
+    /** If the type is a struct, the name of the struct */
+    struct?: string;
+    /** Optional, user-friendly description for the return value */
+    desc?: string;
+  };
+  /** an action is a combination of call/create and an OnComplete */
+  actions: {
+    /** OnCompeltes this method allows when appID === 0 */
+    create: ('NoOp' | 'OptIn' | 'DeleteApplication')[];
+    /** OnCompeltes this method allows when appID !== 0 */
+    call: ('NoOp' | 'OptIn' | 'CloseOut' | 'ClearState' | 'UpdateApplication' | 'DeleteApplication')[];
+  };
+  readonly: boolean;
+}
+
+interface Contract {
+  /** A user-friendly name for the contract */
+  name: string;
+  /** Optional, user-friendly description for the interface */
+  desc?: string;
+  /**
+   * Optional object listing the contract instances across different networks
+   */
+  networks?: {
+    /**
+     * The key is the base64 genesis hash of the network, and the value contains
+     * information about the deployed contract in the network indicated by the
+     * key
+     */
+    [network: string]: {
+      /** The app ID of the deployed contract in this network */
+      appID: number;
```

Comment:
Worth adding in the AlgoKit format as well as another option? i.e. creator address + name

Reply:
I'm trying to think how this could be done in a way that is ARC4 backwards compatible. Don't want to omit appID because that could break an ARC4 client if it tries to use that field (no clients in existence use it that I know of, but just want to be safe)

Reply:
If nothing actually uses this right now then I think it's OK to mess with it and note is as a known deviation from ARC4?

Again, pretty minor feature though.

Reply:
If possible, since we tried to keep ARC4 backward compatibility, I would not pursue this minor deviation. I think declaring the association between `appID` and the Network ID on which the App is deployed has some value.